cmd/webhook-manager/app/server.go

```diff
@@ -77,11 +77,13 @@ func Run(config *options.Config) error {
 
 		klog.V(3).Infof("Registered '%s' as webhook.", service.Path)
 		http.HandleFunc(service.Path, service.Handler)
-
-		klog.V(3).Infof("Registered configuration for webhook <%s>", service.Path)
-		registerWebhookConfig(kubeClient, config, service, config.CaCertData)
 	})
 
+	if err = addCaCertForWebhook(kubeClient, config.CaCertData); err != nil {
+		return fmt.Errorf("failed to add caCert for webhook %v", err)
+	}
```

Comment:
The error seems to be that the webhook server needs to add cacert before it starts, but this function `addCaCertForWebhook` is a time-consuming operation, so scheduler reports the error `connect: connection refused`, so it's better to be done in a goroutine